resources/product.py

```diff
@@ -40,66 +44,87 @@
 class Product(Resource):
     """CRUD for accessing/manipulating a single product document"""
 
-    @staticmethod
-    def get(product_id):
+    # GET - http://127.0.0.1:5000/api/product/<string:product_id>
+    @classmethod
+    @exception_handler
+    def get(cls, product_id):
         """Handles get request for retrieving a single product"""
         product = product_model.objects(_id=product_id).first()
         if len(product) == 0:
             abort(HTTPStatus.NOT_FOUND, message="Product could not be created")
         return json.loads(product.to_json()), HTTPStatus.CREATED
 
-    @staticmethod
+    # POST - http://127.0.0.1:5000/api/product
+    @classmethod
     @exception_handler
     @authenticated
-    def post():
+    def post(cls):
         """Handles the post request and creates a new product"""
-        product = product_args.parse_args()
-        new_product = product_model(**product)
+        body = clean_arguments(product_args)
+        product_id = ObjectId()
+        new_product = product_model(**body, _id=product_id)
+        new_group = group_model(
+            product_id=product_id, organizer_id=body["user_id"], status="Funding"
+        )
         try:
             new_product.save()
+            new_group.save()
+
         except ValidationError as validation_error:
             raise MongoErrorHandler(
-                ValidationError.to_dict(),  # pylint: disable = no-value-for-parameter
-                HTTPStatus.BAD_REQUEST,
+                validation_error.to_dict(), HTTPStatus.BAD_REQUEST
             ) from validation_error
         return json.loads(new_product.to_json()), HTTPStatus.OK
 
-    @staticmethod
-    # @exception_handler
-    # @authenticated
-    def put(product_id):
+    # PUT - http://127.0.0.1:5000/api/product/<string:product_id>
+    @classmethod
+    @exception_handler
+    @authenticated
+    def put(cls, product_id):
         """Handles the put request to update a single product"""
         body = {}
         for key, value in product_args.parse_args().items():
             if value is not None:
                 body[key] = value
```

Comment:
Replace with the `clean_arguments` method

```suggestion
        body = clean_arguments(product_args)
```

Reply:
Will fixed on next PR